BasicSteps/ProcessStep.cs

```diff
@@ -73,9 +84,28 @@ public Int32 Timeout
         ManualResetEvent outputWaitHandle, errorWaitHandle;
         StringBuilder output;
 
+        public ProcessStep()
+        {
+            Rules.Add(HasNoDuplicateEnvironmentVariables, "Cannot have multiple environment variables with the same name.", nameof(EnvironmentVariables));
+        }
+
+        private bool HasNoDuplicateEnvironmentVariables()
+        {
+            HashSet<string> seenVariables = new HashSet<string>();
+            foreach (var variable in EnvironmentVariables)
+            {
+                if (seenVariables.Contains(variable.Name))
+                {
+                    return false;
+                }
+                seenVariables.Add(variable.Name);
```

Comment:
HashSet.Add returns false if the value already exists in the collection, maybe that can be used here?

Reply:
Yup, that is totally possible, changed in the latest commit.